DirectProgramming/C++SYCL_FPGA/Tutorials/Features/state_machine/src/state_machine.hpp

```diff
@@ -0,0 +1,129 @@
+#ifndef STATE_MACHINE_HPP
+#define STATE_MACHINE_HPP
+
+#include <sycl/ext/intel/experimental/task_sequence.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/pipes_ext.hpp>
+#include <sycl/sycl.hpp>
+
+enum State {
+  PROCESS,
+  LD_COEFF
+};
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class IDNaiveStateMachine;
+class IDProperStateMachine;
+class IDStreamDataIn_NaiveStateMachine;
+class IDStreamDataIn_ProperStateMachine;
+class IDStreamDataOut_NaiveStateMachine;
+class IDStreamDataOut_ProperStateMachine;
+
+// define data pipes
+using StreamingPipeProps = decltype(sycl::ext::oneapi::experimental::properties(
+    sycl::ext::intel::experimental::uses_valid<true>,
+    sycl::ext::intel::experimental::bits_per_symbol<32>,
+    sycl::ext::intel::experimental::protocol<
+        sycl::ext::intel::experimental::protocol_name::
+            avalon_streaming_uses_ready>));
+using MyStreamingBeat =
+    sycl::ext::intel::experimental::StreamingBeat<float, true, false>;
+
+using StreamIn_NaiveStateMachine =
+    sycl::ext::intel::experimental::pipe<IDStreamDataIn_NaiveStateMachine, MyStreamingBeat, 0,
+                                         StreamingPipeProps>;
+using StreamIn_ProperStateMachine =
+    sycl::ext::intel::experimental::pipe<IDStreamDataIn_ProperStateMachine, MyStreamingBeat, 0,
+                                         StreamingPipeProps>; 
+using StreamOut_NaiveStateMachine =
+    sycl::ext::intel::experimental::pipe<IDStreamDataOut_NaiveStateMachine, MyStreamingBeat, 0,
+                                         StreamingPipeProps>;
+using StreamOut_ProperStateMachine =
+    sycl::ext::intel::experimental::pipe<IDStreamDataOut_ProperStateMachine, MyStreamingBeat, 0,
+                                         StreamingPipeProps>;
+
+// function for task_sequence
+float updateCoeff(float *coeffPtr, int *executionCount) {
+  float coeffNew = coeffPtr[0];
```

Comment:
```suggestion
  float coeff_new = coeff_ptr[0];
```
Variable names should be snake_case
Function names should be PascalCase()

Variable names should be `snake_case`
Function names should be `PascalCase()`
constants should be `kCamelCase` (the leading `k` is important)